setuptools/tests/test_easy_install.py

```diff
@@ -744,6 +745,49 @@ def test_setup_requires_with_python_requires(self, monkeypatch, tmpdir):
         eggs = list(map(str, pkg_resources.find_distributions(os.path.join(test_pkg, '.eggs'))))
         assert eggs == ['dep 1.0']
 
+    def test_setup_requires_with_transitive_extra_dependency(self, monkeypatch):
```

Comment:
The addition of this test seems very relevant 😃   
You should probably add another (very similar) to test the same behaviour in `tests_require`, or maybe modify your existing function to do the 2 situations in 1 function.

Reply:
Good point.

Reply:
Yeah, but what all those `test_setup_requires...` tests really test is implementation of `fetch_build_egg`.